content/how-to/code-splitting/splitting-vendor.md

```diff
@@ -0,0 +1,132 @@
+
+This is not ideal for the application. If the code in `index.js` changes, then the whole bundle is rebuilt. The browser will have to load a new copy of the new bundle even though most of it hasn't changed at all.
+
+### Multiple entries
+
+Let's try to mitigate this by adding a separate entry point for `moment` and name it `vendor`
+
+```javascript
+\\ webpack.config.js
+
+modules.export = function(env) {
+    return {
+        entry: {
+            main: './index.js',
+            vendor: 'moment'
+        },
+        output: {
+            filename: '[chunkhash].[name].js`,
+            path: './dist'
+        }
+    }
+}
+```
+
+On running `webpack` now, we see that two bundles have been created. If you inspect these though, you will find that the code for `moment` is present in both the files!
+
+It is for this reason, that we will need to use the [CommonsChunkPlugin](https://github.com/webpack/webpack/blob/master/lib/optimize/CommonsChunkPlugin.js).
+
+### CommonsChunkPlugin
+
+This is a pretty complex plugin. It fundamentally allows us to extract all the common modules from different bundles and add them to the common bundle. If a common bundle does not exist, then it creates a new one.
+
+We can modify our webpack config file to use the CommonsChunkPlugin as follows
+
+```javascript
+\\ webpack.config.js
+
+var webpack = require('webpack');
+modules.export = function(env) {
+    return {
+        entry: {
+            main: './index.js',
+            vendor: 'moment'
+        },
+        output: {
+            filename: '[chunkhash].[name].js`,
+            path: './dist'
+        },
+        plugins: [
+            new webpack.optimize.CommonsChunkPlugin({
+                name: 'vendor' // Specify the common bundle's name.
+            })
+        ]
+    }
+}
+```
+Now run `webpack` on your application. Bundle inspection shows that `moment` code is present only in the vendor bundle.
+
+### Manifest file
+
+But, if we change application code and run `webpack` again, we see that the hash for the vendor file changes. Even though we achieved separate bundles for `vendor` and `main` bundles, we see that the `vendor` bundle changes when the application code changes.
+This means that we still don't reap the benefits of browser caching because the hash for vendor file changes on every build and the browser will have to reload the file.
+
+The issue here is that on every build, webpack generates some webpack runtime code, which helps webpack do it's job. When there is a single bundle, the runtime code resides in it. But when multiple bundles are generated, the runtime code is extracted into the common module, here the `vendor` file.
+
+To prevent this, we need extract out the runtime into a separate manifest file. Even though we are creating another bundle, the overhead is offset by the long term caching benefits that we obtain on the `vendor` file.
+
+```javascript
+\\ webpack.config.js
```

Comment:
Push filenames outside of the examples so you get something like this:

**webpack.config.js**

```javascript
demo goes here
```